scripts/exregional_run_prepstart.sh

```diff
@@ -241,28 +241,47 @@ else
      done
   fi
 #
+  filelistn="fv_core.res.tile1.nc fv_srf_wnd.res.tile1.nc fv_tracer.res.tile1.nc phy_data.nc sfc_data.nc"
   checkfile=${bkpath}/${restart_prefix}coupler.res
+  n_iolayouty=$(($IO_LAYOUT_Y-1))
+  list_iolayout=$(seq 0 $n_iolayouty)
   if [ -r "${checkfile}" ] ; then
-    cp_vrfy ${bkpath}/${restart_prefix}fv_core.res.tile1.nc       fv_core.res.tile1.nc 
-    cp_vrfy ${bkpath}/${restart_prefix}fv_tracer.res.tile1.nc     fv_tracer.res.tile1.nc
-    cp_vrfy ${bkpath}/${restart_prefix}sfc_data.nc                sfc_data.nc 
     cp_vrfy ${bkpath}/${restart_prefix}coupler.res                coupler.res
     cp_vrfy ${bkpath}/${restart_prefix}fv_core.res.nc             fv_core.res.nc
-    cp_vrfy ${bkpath}/${restart_prefix}fv_srf_wnd.res.tile1.nc    fv_srf_wnd.res.tile1.nc
-    cp_vrfy ${bkpath}/${restart_prefix}phy_data.nc                phy_data.nc
+    if [ "${IO_LAYOUT_Y}" == "1" ]; then
+      for file in ${filelistn}; do
+        cp_vrfy ${bkpath}/${restart_prefix}${file}     ${file}
+      done
+    else
+      for file in ${filelistn}; do
+        for ii in $list_iolayout
+        do
+          iii=`printf %4.4i $ii`
+          cp_vrfy ${bkpath}/${restart_prefix}${file}.${iii}     ${file}.${iii}
+        done
+      done
+    fi
     cp_vrfy ${fg_root}/${YYYYMMDDHHmInterv}${SLASH_ENSMEM_SUBDIR}/${fg_restart_dirname}/INPUT/gfs_ctrl.nc  gfs_ctrl.nc
     if [ ${SAVE_CYCLE_LOG} == "TRUE" ] ; then
       echo "${YYYYMMDDHH}(${cycle_type}): warm start at ${current_time} from ${checkfile} " >> ${EXPTDIR}/log.cycles
     fi
 #
 # remove checksum from restart files. Checksum will cause trouble if model initializes from analysis
 #
-    ncatted -a checksum,,d,, fv_core.res.tile1.nc
-    ncatted -a checksum,,d,, fv_tracer.res.tile1.nc
-    ncatted -a checksum,,d,, sfc_data.nc
+    if [ "${IO_LAYOUT_Y}" == "1" ]; then
+      for file in ${filelistn}; do
+        ncatted -a checksum,,d,, ${file}
+      done
+    else
+      for file in ${filelistn}; do
+        for ii in $list_iolayout
+        do
+          iii=`printf %4.4i $ii`
+          ncatted -a checksum,,d,, ${file}.${iii}
```

Comment:
Does the ncatted command take a lot of time to complete? If so, we can make all these ncatted command run in parallel in the future.

Reply:
I did not par attention to run time from this piece of code. The whole prep task takes about 200s now.